askup/utils/views.py

```diff
@@ -305,22 +341,25 @@ def apply_filter_to_queryset(request, filter, queryset):
     return queryset
 
 
-def do_user_checks_and_evaluate(user, answer, evaluation):
-    """Do user checks and evaluate answer for the answer evaluation view."""
+def do_user_checks_and_evaluate(user, answer, evaluation, qset_id):
+    """
+    Do user checks and evaluate answer for the answer evaluation view.
+    """
     evaluation_int = int(evaluation)
     is_admin = check_user_has_groups(user, 'admin')
+    user_permitted = Organization.objects.filter(qset__in=[qset_id], users__in=[user.id]).first()
 
-    if not is_admin and user not in answer.question.qset.top_qset.users.all():
+    if not is_admin and not user_permitted:
         return False
 
-    if evaluation_int in tuple(zip(*Answer.EVALUATIONS))[0] and answer:
+    if evaluation_int in next(zip(*Answer.EVALUATIONS)) and answer:
```

Comment:
I'm curious is it really possible the case when answer == None?

Reply:
Good question, thanks.
Yes, it is.
There's a time gap between user answered the question and he evaluated himself.
In this gap somebody may delete the question and in a cascade way there will be deleted this answer as well.

Reply:
Got it, some comments will be very helpful here.